models/marts/dim_contract_code_hist.sql

```diff
@@ -0,0 +1,242 @@
+{{
+    config(
+        materialized = 'incremental',
+        unique_key = ['ledger_key_hash', 'start_date'],
+        cluster_by = ["ledger_key_hash", "start_date", "row_hash"],
+        tags = ["soroban_analytics", "dimension", "scd-2", "daily"]
+    )
+}}
+
+/*
+Model: dim_contract_code_hist
```

Comment:
nit: name scd2 tables with `_scd2` instead of `_hist`

Reply:
- Opted for _hist to make this follow _current naming format we already have. I am open to suggestions/feedback here